### PR TITLE
Issue 179

### PR DESCRIPTION
**Pull request by GTBoon72**
#179 This PR should fix this issue. Checked other self.name references, but they work fine as is.

### CODE REVIEW

**Comment by FriedrichFroebel** [comment]
Thanks for the PR. Unfortunately your changes fail in the CI which seems to be an issue with the Betamax cassettes being outdated and requiring an update as well.

**Comment by FriedrichFroebel** [comment]
The problem is not in the testing itself, but in the Geocaching.com site changes. For this reason your tests fail at the moment and you have to re-record the corresponding cassettes. Otherwise I will not be able to merge your changes.

**Comment by GTBoon72** [comment]
Thanks for pointing out. Will see if I understand how that works, and will give it a try this weekend.

**Comment by FriedrichFroebel** [comment]
The PM-only property can only be detected if you are logged in as a basic user, which is implicitly documented in the `load_by_guid` method, although this might not be obvious enough.

For basic members the print page of a PM-only cache will be mostly empty and contain the following snippet:

```html
        <div id="pnlErrorMessage">
	<p class="Warning">You must be a Premium Member to view this page.</p>
</div>
```

For non PM-only caches it will look like this:

```html
        <div id="pnlErrorMessage">

</div>
```

I will have a look at updating the test cassettes later. If you are okay with it and you have allowed reviewers to make changes to your pull request (and I manage to get this to work on my side), I can update the corresponding cassettes on your branch for you to check if your changes are okay or need a fix.

**Comment by GTBoon72** [comment]
Hmm, yeah, that makes sense. But then, am I supposed to switch between PM and non-PM accounts, while running the tests? Did I overlook that in the test description?
Of course I am okay with your updates, it's your project after all! I enabled reviewers to update the code.
Thanks for your help!

**Comment by tomasbedrich** [comment]
I am leaving that to you @FriedrichFroebel. Please ping me as soon as you need a release. Thanks!

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -82,17 +82,18 @@ If you haven't written or modified any tests, tests can be run just like:
...
 worry, these will not leave your computer. Betamax will insert a placeholder when it records any
 new cassettes. To run new tests, first set up the following environment variables:
 
 .. code-block:: bash
 
-    PYCACHING_TEST_USERNAME="yourusername" PYCACHING_TEST_PASSWORD="yourpassword" pytest
+    PYCACHING_TEST_USERNAME="yourusername" PYCACHING_TEST_PASSWORD="yourpassword" pytest <test folder name>
 
 Substitute your username for ``yourusername`` and your password for ``yourpassword``.
+NB: for successful test results, you'll need to use a Basic Member account!
```
Comment (FriedrichFroebel) [inline CONTRIBUTING.rst:96]: I would rather use something like

> This requires you to use a basic member account, otherwise you might see unexpected test failures.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -82,17 +82,18 @@ If you haven't written or modified any tests, tests can be run just like:
 
 .. code-block:: bash
 
-    pytest
+    pytest <test folder name>
```
Comment (FriedrichFroebel) [inline CONTRIBUTING.rst:85]: Why did you change this? If you are in the root directory of the repository, it should work out of the box without specifying the directory.

Comment (GTBoon72) [reply CONTRIBUTING.rst:85]: At first, that didn't work for me, it didn't execute any tests at all. Now that you mention it, I tried it without a default, and now it does work. Nonetheless, I think it is a useful addition, since it explains how to switch between the tests in the 'test' and 'tests_new' folders.

**Review by FriedrichFroebel — APPROVED** [review]
Thank you for your patience.
